plugins/pulp_rpm/plugins/importers/yum/repomd/metadata.py

```diff
@@ -301,8 +318,11 @@ def generate_dbs(self):
                         db_key = self.generate_db_key(unit_key)
                         db_file_handle[db_key] = raw_xml
                     db_file_handle.sync()
+                    package_count[filename] = len(db_file_handle)
             self.dbs[filename] = db_filename
 
+        return package_count
```

Comment:
I described the reasoning behind those exceptions in other comment. If it does not seem to be valid then it makes sense to compare some counters here.

As for the class attributes, initially they were in this class and not in `RepoSync`.
But if we decide to modify `RepoSync._identify_wanted_versions` and raise exception right there, then we have issue with access to the counters if they are attributes of this class and not the `RepoSync` class.
So if we go with "compare and raise at the moment we have numbers no matter where we are", then +1 for generator wrapper your described earlier, in this case we will be able to have counters as attribute of this class.